### ISSUE TITLE
Bug: (no-dupe-keys) The rule should differentiate computed `__proto__` and non-computed `__proto__`

### ISSUE THREAD

**Issue by JLHwung** [eslint/eslint#19504]
### Environment

Node version: v23.6.0
npm version: v10.9.2
Local ESLint version: v9.13.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.3.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
    {
        "rules": {
            "no-dupe-keys": [
                "error"
            ]
        }
    }
];
```
</details>

```js
({
  ["__proto__"]: null,
  __proto__: null
})
```

### What did you expect to happen?

No errors. The computed `__proto__` key is a valid property while the non-computed one is the prototype setter.

### What actually happened?

3:3 Error: Duplicate key '__proto__'.  ([no-dupe-keys](https://eslint.org/docs/rules/no-dupe-keys))

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiKHtcbiAgW1wiX19wcm90b19fXCJdOiBudWxsLFxuICBfX3Byb3RvX186IG51bGxcbn0pIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby1kdXBlLWtleXMiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Found this issue when drafting a Babel test: https://github.com/babel/babel/pull/17175. We have to disable this rule for the specific line.

**Comment by mdjermanovic** [comment]
Thanks for the report, I can reproduce this bug. I'm working on the fix.
